### PR TITLE
feat(core): Log and meter events when kafka recovery drops data

### PR DESCRIPTION
**Pull request by vishramachandran**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)

System does not log or meter events when kafka offset to recover from is not available.

**New behavior :**

Log and meter events when kafka offset to recover from is not available. Indicates potential data loss.

### CODE REVIEW

File: core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala [hunk 1]
```diff
@@ -145,12 +145,24 @@ extends MemStore with StrictLogging {
                     endOffset: Long,
                     checkpoints: Map[Int, Long],
                     reportingInterval: Long): Observable[Long] = {
+    var startOffsetValidated = false
```
Comment (broneill) [inline core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala]: Just curious: Can this new var be declared within the first scope that needs it?

Comment (vishramachandran) [reply core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala]: done

File: core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala [hunk 2]
```diff
@@ -150,7 +150,19 @@ extends MemStore with StrictLogging {
...
       var targetOffset = startOffset + reportingInterval
-      stream.map(shard.ingest(_)).collect {
+      var startOffsetValidated = false
+      stream.map { r =>
+        if (!startOffsetValidated) {
+          if (r.offset > startOffset) {
+            val offsetsNotRecovered = r.offset - startOffset
+            logger.error(s"Could not recover dataset=$dataset shard=$shardNum from check pointed offset possibly " +
+                         s"because of retention issues. offsetsNotRecovered=$offsetsNotRecovered")
+            shard.shardStats.offsetsNotRecovered.increment(offsetsNotRecovered)
```
Comment (velvia) [inline core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala]: This is fine, though I feel like in the future these should be "events" and not metrics.

Comment (vishramachandran) [reply core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala]: agreed.

**Pull request merged**